_posts/2015-07-23-postgres-review-operators.md

```diff
@@ -0,0 +1,292 @@
+---
+layout:     post
+title:      Review of operators in Postgres
+date:       2015-07-23 23:00:00
+summary:    In this post we will review some operators so we can have
+            a place to reference in future posts.
+categories: sql
+---
+
+Let's investigate some operators in Postgres and how to use them
+in our day-to-day work. The objective is to have a centralized guide
+to refer to when in need.
+
+### Array
+
++ `@>` is the operator that specifies if an Array **contains** another Array.
+
+{% highlight sql %}
+SELECT array[1,3,6] @> array[1,4,7];
+
+ ?column?
+----------
+ f
+(1 row)
+
+SELECT array[1,3,6] @> array[1,3];
+
+ ?column?
+----------
+ t
+(1 row)
+{% endhighlight %}
+
++ `<@` on the other hand specifies if an Array **is contained by** another one.
+
+{% highlight sql %}
+SELECT array[1,3] <@ array[4,5,6];
+
+ ?column?
+----------
+ f
+(1 row)
+
+SELECT array[4,6] <@ array[4,5,6];
+
+ ?column?
+----------
+ t
+(1 row)
+{% endhighlight %}
+
++ `&&` tells if an Array has **elements in common** with another Array.
+
+{% highlight sql %}
+SELECT array[3,6] && array[1,4,7];
+
+ ?column?
+----------
+ f
+(1 row)
+
+SELECT array[3,6] && array[1,3,6];
```

Comment:
I would use `SELECT array[2,3,6] && array[1,3,6];` just to explicit the answer will be `true` if I use an element (number 2) that isn't there but the others are (number 3 and 6).